test/modules/DirectMethodSender/DirectMethodSenderBase.cs

```diff
@@ -37,7 +37,30 @@ abstract class DirectMethodSenderBase : IDisposable
             logger.LogInformation($"{this.GetType().ToString()} : Calling Direct Method on device {this.deviceId} targeting module [{this.targetModuleId}] with count {this.directMethodCount}.");
             try
             {
-                int resultStatus = await this.InvokeDeviceMethodAsync(this.deviceId, this.targetModuleId, methodName, this.directMethodCount, CancellationToken.None);
+                const int maxRetry = 3;
+                int resultStatus = (int)HttpStatusCode.InternalServerError;
+                int transientRetryCount = 0;
+
+                while (transientRetryCount <= maxRetry)
+                {
+                    try
+                    {
+                        transientRetryCount++;
+                        resultStatus = await this.InvokeDeviceMethodAsync(this.deviceId, this.targetModuleId, methodName, this.directMethodCount, CancellationToken.None);
+                    }
+                    catch (IotHubCommunicationException e) when (e.IsTransient)
+                    {
+                        if (transientRetryCount < maxRetry)
+                        {
+                            logger.LogInformation(e, $"Transient IotHubCommunicationException caught with count {this.directMethodCount}.");
+                        }
+                        else
+                        {
+                            logger.LogInformation(e, $"Transient IotHubCommunicationException caught with count {this.directMethodCount}. Retry Exceeded.");
+                            return new Tuple<HttpStatusCode, ulong>(HttpStatusCode.RequestTimeout, this.directMethodCount);
```

Comment:
Remove this line and just set resultStatus  = HttpStatusCode.RequestTimeout inside catch block.